### DISCUSSION TITLE
Reporting in on 2 issues

### DISCUSSION THREAD

**Discussion by SpeedyCharly** [AzuraCast/AzuraCast#4963] [Q&A]
Was scanning my logs for still ever-present hacker activity for which I have since created a suggestion. 

At the same time I noted that buffer overruns had returned to the fore over the last 2 days and than CPU activity was again going through the roof. Not quite enough to freeze the stream as before, but still of concern. 
[Hack.txt](https://github.com/AzuraCast/AzuraCast/files/7832116/Hack.txt)

I was on Rolling release #683ddf at the time and proceeded to update to #abaa4a0 which returned CPU and buffer to normal right away. I am however still seeing a lot of instances like this;

2022/01/07 12:00:19 [amplify_0:4] Overriding amplification: 1.000000.
2022/01/07 12:00:19 [amplify_2:4] Overriding amplification: 1.000000.
2022/01/07 12:00:19 [process:4] Starting process
2022/01/07 12:00:19 [process:4] Closing process's stdin
2022/01/07 12:00:19 [process:4] Process exited with code 0
2022/01/07 12:00:19 [process:4] Cleaning up process
2022/01/07 12:00:19 [lang:3] AzuraCast Feedback Response: OK
2022/01/07 12:00:20 [buffer_417:4] Buffer emptied, buffering needed.
2022/01/07 12:00:20 [buffer_416:4] Buffer emptied, buffering needed.
2022/01/07 12:00:20 [crossfade_0:4] Buffering end of track...
2022/01/07 12:00:21 [crossfade_0:4] More buffering will be needed.
2022/01/07 12:00:36 [input_streamer:4] End of track.
2022/01/07 12:00:36 [live_fallback:3] Switch to map_metadata_5 with transition.
2022/01/07 12:00:36 [amplify_0:4] End of the current overriding.
2022/01/07 12:00:36 [amplify_2:4] End of the current overriding.

Does that mean that the buffering issue isn't quite resolved yet?

**Comment by ItsMitchh** [comment]
In the logs, the error 401 is for unauthenicated users, meaning they could not authenticate. I have done some reviews of the IPs you have provided and they trace back to Censys.io. It's essentially a bot / automated crawler script which checks commonly used ports to see if they're open to see if there's any security vulnerabilities. As per their written press report - https://support.censys.io/hc/en-us/articles/360043177092-Opt-Out-of-Scanning

**Reply by SpeedyCharly** [reply]
Yes I was quite aware of the meaning of the 401 error - what I don't get is that it appears to be in response to our operator djPhiloup who was live at the time any didn't change his credentials on the fly.  The reference to the questionnable ip appears after the 401 error...

edit; not a Linuxian geek here; I have no idea how to configure the server's firewall

**Reply by ItsMitchh** [reply]
You can use this guide to block IPs from accessing your server. 

https://www.cyberciti.biz/faq/how-to-block-an-ip-address-with-ufw-on-ubuntu-linux-server/

**Reply by SpeedyCharly** [reply]
Danke.... will try that

**Reply by SpeedyCharly** [reply]
But in any case, I would understand such bots doing their thing on us once a month or so, but not multiple times every day...

**Reply by SpeedyCharly** [reply]
> You can use this guide to block IPs from accessing your server.
> 
> https://www.cyberciti.biz/faq/how-to-block-an-ip-address-with-ufw-on-ubuntu-linux-server/

A caveat on using ufw to block nuisance IPs ; if you don't quite know the usage protocols DO NOT - Repeat DO NOT - activate ufw on your server - you may well block yourself out of your server or VPS altogether.... just learned that the hard way!!!  And, recovering from such lockout is a bitch if you no longer have direct SSH access to your server.

Please investigate the NOMOREDDOS avenue if you need to block IPs... your server or VPS will love you for it!!!!

**Reply by BusterNeece** [reply]
@SpeedyCharly Yeah, a key thing is to make sure SSH is cleared through UFW first (`ufw allow ssh`) as a matter of priority. Otherwise, you'll have to get in through the recovery console.

**Reply by SpeedyCharly** [reply]
Yes... exactly what happened to me cuz the help file I was following never said sh*t about that part...  I also later found warnings about getting locked out of tcp as well...  Had to put my VPS in recovery mode, access the system with temporary root access, learn me how to mount drives and chroot to VPS so that I could disable ufw and restart the server properly...

A bit of grief... but I learned a bit more Linuxian lingo   :P

**Comment by SpeedyCharly** [comment]
Would any one care to respond to the concern I expressed over the "more buffering will be needed" thing?

**Reply by BusterNeece** [reply]
That's a question for Liquidsoap.

**Reply by SpeedyCharly** [reply]
Merci beaucoup!  Shot the question over to their side of the fence... :P

**Comment by SpeedyCharly** [comment]
[toots
1 hour ago
Maintainer
It looks like something is blocking the main streaming thread when a user connects. Is there an API call happening on login?]

Toots suggests debugging AzuraCast API calls on DJ login in an effort to resolve this buffer thing... I have no idea what that means...

**Reply by BusterNeece** [reply]
There's certainly an API call to "authenticate" the DJ.

**Reply by SpeedyCharly** [reply]
Ok then, I guess Toots would want us to check if all is ok at that end before proceeding further...

**Reply by BusterNeece** [reply]
I suppose the question would be...well, that's not going to be an INSTANT response, so if it delays, it shouldn't block in a way that makes any part of the stream halt. That's something we'd like to be able to work around on our end, otherwise dynamic DJ authentication is always going to be difficult.

**Reply by SpeedyCharly** [reply]
I have set a 30 second buffer in Azura to help DJs login on first attempt without timing out.  But I'm wondering why the first handshake is to port 8006 when our encoder calls for 8005.  We are on ShoutcastV1 because some of our guys use Virtual Dj, but I don't see the reason behind going to 8006, then handing over to 8005 after.  Further, why is the dj authenticated at every change of media?  There should be no need unless another IP tries to log in...

**Reply by SpeedyCharly** [reply]
Here is a portion of log showing our first DJ of the day logging in if it's any help :
[First Login.txt](https://github.com/AzuraCast/AzuraCast/files/7836810/First.Login.txt)

**Comment by SpeedyCharly** [comment]
Cross-reference with; https://github.com/AzuraCast/AzuraCast/issues/4983 and https://github.com/savonet/liquidsoap/issues/2129

**Comment by SpeedyCharly** [comment]
Again receiving log-in calls from nuisance IPs;
[US IP calling.txt](https://github.com/AzuraCast/AzuraCast/files/7910857/US.IP.calling.txt)

I for one am not interested in spending my waking hours reading logs and manually blocking IPs. We definitely need a solution to be able to whitelist IPs accessing incoming streaming ports and deny all others right at the front door...

**Reply by ItsMitchh** [reply]
Features and suggestions can be made at https://features.azuracast.com
